### PR TITLE
Add Rpc metadata for Slice items

### PR DESCRIPTION
**Pull request by pepone**
This PR adds a Rpc metadata for Slice items that corresponds to the slicec-cs rpc parameter.

### CODE REVIEW

File: tools/IceRpc.Slice.Tools/IceRpc.Slice.Tools.props [hunk 1]
```diff
@@ -49,6 +49,7 @@
   <ItemDefinitionGroup>
     <SliceFile>
       <OutputDir>$(MSBuildProjectDirectory)/generated</OutputDir>
+      <Rpc>icerpc</Rpc>
```
Comment (bernardnormier) [inline tools/IceRpc.Slice.Tools/IceRpc.Slice.Tools.props:52]: This ItemDefinitionsGroup is where we specify the default values?

File: tools/IceRpc.Slice.Tools/README.md [hunk 2]
```diff
@@ -84,12 +84,13 @@ The [ZeroC.Slice][zeroc-slice] NuGet package follows this pattern and provides d
...
+| Name              | Default   | Description                                                                                                                         |
+|-------------------|-----------|-------------------------------------------------------------------------------------------------------------------------------------|
+| AdditionalOptions |           | Specifies additional options to pass to [`slicec-cs`][slicec-cs].                                                                   |
+| OutputDir         | generated | Sets the output directory for the generated code. This metadata corresponds to the `--output-dir` option of `slicec-cs`.            |
+| Pack              | `false`   | Specifies whether or not to include the items (Slice files) in the NuGet package.                                                   |
+| PackagePath       | slice     | Sets the target path in the NuGet package. Used only when Pack is `true`.                                                           |
+| Rpc               | icerpc    | Specifies the RPC framework to generate code for `icerpc` or `none`. This metadata corresponds to the `--rpc` option of `slicec-cs` |
```
Comment (bernardnormier) [inline tools/IceRpc.Slice.Tools/README.md:93]: This does not sound correct. What about:
Specifies the RPC framework to generate code for. Possible values: `icerpc`, `none`. This metadata corresponds...

File: tools/IceRpc.Slice.Tools/SliceCCSharpTask.cs [hunk 3]
```diff
@@ -64,6 +70,21 @@ public class SliceCCSharpTask : ToolTask
 
     private readonly JsonSerializerOptions _jsonSerializeOptions = new() { PropertyNameCaseInsensitive = true };
 
+    protected override bool ValidateParameters()
+    {
+        var validRpcValues = new[] { "icerpc", "none" };
+
+        if (!validRpcValues.Contains(Rpc, StringComparer.OrdinalIgnoreCase))
+        {
+            Log.LogError($"Invalid Rpc value '{Rpc}'. Valid options are 'icerpc' and 'none'.");
```
Comment (bernardnormier) [inline tools/IceRpc.Slice.Tools/SliceCCSharpTask.cs:79]: Should be Valid values are...

File: tools/IceRpc.Slice.Tools/SliceCCSharpTask.cs [hunk 4]
```diff
@@ -26,6 +26,12 @@ public class SliceCCSharpTask : ToolTask
     /// to <c>-R</c> slicec-cs compiler option.</summary>
     public string[] References { get; set; } = Array.Empty<string>();
 
+    /// <summary>
+    /// The RPC provider to generate code for, corresponds to the <c>--rpc</c> slicec-cs compiler option.
+    /// </summary>
+    [Required]
+    public string Rpc { get; set; } = "";
```
Comment (bernardnormier) [inline tools/IceRpc.Slice.Tools/SliceCCSharpTask.cs:33]: The `[Required]` attribute does not suppress the need to have a default value for this property?

File: tools/IceRpc.Slice.Tools/SliceFile.FileProperties.xaml [hunk 5]
```diff
@@ -85,6 +85,15 @@
         </StringProperty.DataSource>
     </StringProperty>
 
+    <EnumProperty Name="Rpc"
+                  DisplayName="Rpc Provider"
```
Comment (bernardnormier) [inline tools/IceRpc.Slice.Tools/SliceFile.FileProperties.xaml:89]: I would use RPC Provider for the display name

File: tools/IceRpc.Slice.Tools/SliceFile.ProjectProperties.xaml [hunk 6]
```diff
@@ -33,6 +33,15 @@
         </StringProperty.DataSource>
     </StringProperty>
 
+    <EnumProperty Name="Rpc"
+                  DisplayName="Rpc Provider"
```
Comment (bernardnormier) [inline tools/IceRpc.Slice.Tools/SliceFile.ProjectProperties.xaml:37]: Same here.

Also, do we need in both SliceFile.FileProperties.xaml and SliceFile.ProjectProperties.xaml?

**Review by bernardnormier — APPROVED** [review]

File: tools/IceRpc.Slice.Tools/SliceFile.ProjectProperties.xaml (hunk 6 above)
Comment (pepone) [reply tools/IceRpc.Slice.Tools/SliceFile.ProjectProperties.xaml:37]: One is per file, and the other is per project. This is to allow setting things from Visual Studio UI

File: tools/IceRpc.Slice.Tools/IceRpc.Slice.Tools.props (hunk 1 above)
Comment (pepone) [reply tools/IceRpc.Slice.Tools/IceRpc.Slice.Tools.props:52]: Yes these are the defaults for all SliceFile items

File: tools/IceRpc.Slice.Tools/SliceCCSharpTask.cs (hunk 4 above)
Comment (pepone) [reply tools/IceRpc.Slice.Tools/SliceCCSharpTask.cs:33]: No, this only cause an error if the task caller doesn't provide the parameter.

Without the value I get:

```
warning CS8618: Non-nullable property 'Rpc' must contain a non-null value when exiting constructor
```

**Pull request merged**